test/k8sT/Nightly.go

```diff
@@ -396,6 +396,39 @@ var _ = Describe("NightlyExamples", func() {
 		})
 
 	})
+	Context("Upgrade test", func() {
+		var cleanupCallback = func() { return }
+
+		BeforeAll(func() {
+			kubectl.Exec("sudo docker rmi cilium/cilium")
+		})
+
+		BeforeEach(func() {
+			// Making sure that we deleted the  cilium ds. No assert message
+			// because maybe is not present
+			kubectl.DeleteResource("ds", fmt.Sprintf("-n %s cilium", helpers.KubeSystemNamespace))
+
+			// Delete kube-dns because if not will be a restore the old endpoints
+			// from master instead of create the new ones.
+			_ = kubectl.DeleteResource(
+				"deploy", fmt.Sprintf("-n %s kube-dns", helpers.KubeSystemNamespace))
+
+			ExpectAllPodsTerminated(kubectl)
+		})
+
+		AfterEach(func() {
+			cleanupCallback()
+		})
+
+		It("Update Cilium from 1.0 to master", func() {
+			helpers.InstallExampleCilium(kubectl, "cilium/cilium:v1.0.4")
```

Comment:
What would be cool (this could be a follow-up) is if we get the contents of the `VERSION` file from the branches that we want.

Reply:
The reason that we have a stale image and no a v1.0 is because this is a separate test, that test the v1.0 upgrade. I would not like to have multiple values in const package.

Reply:
@eloycoto I don't see how that answers Andre's question. We are already at v1.0.5 for Cilium v1.0. As a result, this code is already out of date. I think this PR should test upgrades from various versions with logic outlined in https://github.com/cilium/cilium/issues/5060 .